\begin{abstract}
  Abiotic stresses are a main cause of extensive agricultural
  production losses worldwide.  This paper proposes a workflow to
  identify stress responsive genes in organisms: on input RNA
  sequencing read counts measured for genotypes under control and
  treatment conditions, and biological replicates, it outputs a
  collection of characterized genes, potentially relevant to
  treatment. Technically, the proposed approach is both a
  generalization and an extension of WGCNA. It is showcased with a
  systematic study on rice (\textit{Oryza sativa}), a major food
  source that is known to be highly sensitive to salt stress.  A total
  of 6 modules are detected as relevant in the response to salt stress
  in rice: 3 modules of 3 genes each, all associated with shoot K
  content, 2 modules of 3 genes associated with shoot biomass, and 1
  module of 4 genes associated with root biomass. These genes may act
  as potential targets for the improvement of salinity tolerance in
  rice cultivars.
  %% \keywords{First keyword  \and Second keyword \and Another keyword.}
\end{abstract}
